### PR TITLE
Added multiple network support

### PR DESCRIPTION
**Pull request by thekoma**
As subject

### CODE REVIEW

**Comment by ares** [comment]
This would be great addition. Does it keep the configuration even after installer reboot? I used to have similar patch, but the interfaces file was always rewritten just before reboot. Also would it be possible to add support for bonds to keep feature parity with kickstart snippet? You can see how we deal with bonds at https://github.com/theforeman/community-templates/blob/develop/snippets/kickstart_networking_setup.erb#L33

File: preseed/finish.erb [hunk 1]
```diff
@@ -59,5 +59,5 @@ salt-call --no-color --grains >/dev/null
 <% if chef_enabled %>
 <%= respond_to?(:chef_bootstrap) ? chef_bootstrap(@host) : snippet_if_exists("chef-client omnibus bootstrap") %>
 <% end%>
-
+<%= snippet 'preseed_network' %>
```
Comment (ares) [inline preseed/finish.erb:62]: could you please change the name to "preseed_networking_setup" for consistency?

File: preseed/preseed_network [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#<%= interface %>
+auto $real
+iface $real inet <%= dhcp ? 'dhcp' : 'static' %>
+<% unless dhcp -%>
+    address <%= interface.ip %>
+    netmask <%= subnet.mask %>
+<% end %>
+EOF
+<% end -%>
+<%# service networking restart #not useful in installation %>
```
Comment (ares) [inline preseed/preseed_network:45]: please remove the comment

File: preseed/preseed_network [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+<% vlan = virtual && subnet.has_vlanid? -%>
+<% alias_type = virtual && !subnet.has_vlanid? && interface.identifier.include?(':') -%>
+<% dhcp = subnet.nil? ? false : subnet.dhcp_boot_mode? -%>
+real=`ip -o link | grep <%= interface.mac -%> | awk '{print $2;}' | sed s/:$//`
+<% if virtual -%>
+real=`echo <%= interface.identifier -%> | sed s/<%= interface.physical_device -%>/$real/`
+<% end -%>
+
+cat << EOF >> /etc/network/interfaces
+#<%= interface %>
```
Comment (ares) [inline preseed/preseed_network:36]: probably would be safer to use <%= interface.identifier %>

File: preseed/preseed_network [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<% subnet = @host.subnet -%>
+<% dhcp = subnet.dhcp_boot_mode? -%>
+
+real=`ip -o link | grep <%= @host.mac -%> | awk '{print $2;}' | sed s/://`
+cat << EOF > /etc/network/interfaces
+#LoopBack
+auto lo
+iface lo inet loopback
+
+#<%= @host.primary_interface %>
```
Comment (ares) [inline preseed/preseed_network:10]: probably would be safer to use <%= @host.primary_interface.identifier %>

File: preseed/finish.erb (hunk 1 above)
Comment (thekoma) [inline preseed/finish.erb:62]: Sure!

File: preseed/preseed_network (hunk 2 above)
Comment (thekoma) [inline preseed/preseed_network:45]: Actually the comment is placed in the file exactly to identify the interfzce after creation

File: preseed/preseed_network (hunk 3 above)
Comment (thekoma) [inline preseed/preseed_network:36]: Ares this is exactly your code manipulated to work with new variables. Feel free to make changes. As it is now it works for me.

**Comment by mmoll** [comment]
@koma85 Thanks a bunch for working on that! could you address @ares's comments? Also, the snippet needs a .erb suffix.

**Comment by mmoll** [comment]
@koma85 ping?

**Comment by thekoma** [comment]
It's in my to do list but I'm very busy ATM.
Actually next weekend I've a new big deploy on our laboratory so I'll try to complete the patch.

**Comment by mmoll** [comment]
no problem :ice_cream:

I just didn't want this to get lost.

**Comment by mmoll** [comment]
continuation in #234

**Pull request closed without merging**